src/Domain/BoundaryConditions/BoundaryCondition.hpp

```diff
@@ -0,0 +1,34 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <charm++.h>
```

Comment:
`CkMigrateMessage` is declared in `<pup.h>`, so no need to include `charm++.h`. In fact, I'm considering to create a `Charmxx::pup` cmake target just for linking Charm's serialization framework so not all of our code has to link the entirety of Charm.